### ISSUE TITLE
After Rufus V4,6 (2008) run it reports "Cant assign a drive letter

### ISSUE THREAD

**Issue by Sixhackers** [pbatard/rufus#2606]
﻿

Checklist
---------
- [ ] I have been made aware that if my problem can be summarized as _"I've created or tried to create a media using Rufus, and ..."_, and I am not including a log, **this issue will be summarily closed**.
- [x] I looked at https://github.com/pbatard/rufus/wiki/FAQ to see if my question has already been answered.
- [x] I performed a search in the issue tracker for similar issues using keywords relevant to my problem, such as the error message I got from the log.
- [x] I clicked the 'Log' button (🗒️) or pressed <kbd>Ctrl</kbd>-<kbd>L</kbd> in Rufus, or used [DebugView](https://learn.microsoft.com/en-us/sysinternals/downloads/debugview), and copy/pasted the log into the section that says `<FULL LOG>` below.
- [x] The log I am copying is the FULL log, starting with the line `Rufus version: x.y.z` - I have NOT removed any part of it.

Additionally (if applicable):
- [x] I ran a bad blocks check, by clicking _Show advanced format options_ then _Check device for bad blocks_, and confirmed that my USB is not defective.
- [ ] I also tried one or more of the following:
  - [ ] Using a different USB drive.
  - [ ] Plugging the USB into a different port.
  - [ ] Running Rufus on a different computer.
- [ ] If using an image, I clicked on the `(✓)` button to compute the MD5, SHA1 and SHA256 checksums, which are therefore present in the log I copied. I confirmed, by performing an internet search, that these values match the ones from the official image.

Issue description
-----------------
<Please describe your issue here>When I run rufus after I dismis the warnings rufus reports "unable to create a drive letter" 

Log
---
```
Rufus x64 v4.6.2208
Windows version: Windows 10 Home x64 (Build 19045.5073)
Syslinux versions: 4.07/2013-07-25, 6.04/pre1
Grub versions: 0.4.6a, 2.12
System locale ID: 0x0409 (en-US)
Will use default UI locale 0x0409
SetLGP: Successfully set NoDriveTypeAutorun policy to 0x0000009E
Localization set to 'en-US'
Found 517 revoked UEFI bootloaders from embedded list
Found USB 3.0 device 'Generic STORAGE DEVICE USB Device' (05E3:0749)
1 device found
Disk type: Removable, Disk size: 32 GB, Sector size: 512 bytes
Cylinders: 3891, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x5C5F7054
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: NTFS (0x07)
  Detected File System: NTFS
  Size: 29.8 GB (32004532224 bytes)
  Start Sector: 63, Boot: Yes
Found 5 additional UEFI revocation filters from remote SBAT
Scanning image...
ISO analysis:
  Image is a UDF image
Disk image analysis:
  Image does not have a Boot Marker
ISO label: 'CCCOMA_X64FRE_EN-US_DV9'
  Detected: Windows 11 ISO (Build 26100.1742)
  Size: 5.4 GB (Projected)
  Has a >4GB file
  Uses: EFI
  Uses: Bootmgr (BIOS and UEFI)
  Uses: Install.wim (version 0.13.1)
Using image: Win11_24H2_English_x64.iso (5.4 GB)
Selected Windows User Experience options:
• Bypass SB/TPM/RAM
• Bypass online account requirement
Timeout while retrieving conflicting process list

Format operation started
Requesting disk access...
ABORTED: Cannot use an image that is located on the target drive!
Unable to find a drive letter to use

Found USB 3.0 device 'Generic STORAGE DEVICE USB Device' (05E3:0749)
1 device found
Disk type: Removable, Disk size: 32 GB, Sector size: 512 bytes
Cylinders: 3891, Tracks per cylinder: 255, Sectors per track: 63
Partition type: MBR, NB Partitions: 1
Disk ID: 0x5C5F7054
Drive has a DOS/NT/95A (F2) Master Boot Record
Partition 1:
  Type: NTFS (0x07)
  Detected File System: NTFS
  Size: 29.8 GB (32004532224 bytes)
  Start Sector: 63, Boot: Yes
```

**Comment by Sixhackers** [comment]
I did a fresh download of the Windows 11 ISO file as it warned that the link is only valid for 24 hour. and did try multiple USB ports for the USB ports used to see the Windows 11 ISO file. But it never destroys the data on the USB dive or creates a new drive letter file.

**Comment by pbatard** [comment]
```
ABORTED: Cannot use an image that is located on the target drive!
```

Your ISO is on the same drive as the one you are trying to format. **THIS WILL NEVER WORK** since Rufus needs to **completely erase all data** from that drive before it can start processing the ISO.

Move your ISO outside the destination drive.

**Comment by Sixhackers** [comment]
Hi Pete, Thanks for your quick reply.   The Win 11 ISO image to be used fot the upgrade is on an external 32GB drive. The log shows this as "Found USB 3.0 device 'Generic STORAGE DEVICE USB Device' (05E3:0749). However I did originally download the Win 11 ISO image 1st onto the C: and then copied it to the external flash drive. So I guess I'l try to delete the original D/L from the C: and try again. However when using the "Select" button I do select the Win 11 ISO image thats on the external drive and the app says ready. But as stated once Rufus starts and I dismiss the warning about multiple partitions on the C:, the next message is "unable to assign a drive letter" I know you've closed the report but I will try errasing the original D/L of the Win 11 image from the C: even though the "Select" button directed the Image to be used was on the external drive. Note: I did also finally try modifying the registry (Option 1)  and had the same error. So  I changed the registry back. I'l delete that original copy of the Win 11 ISO from the original D/L storage place on the main C: and see if it is the cause. I'l report back in comments. Thanks for such a great project!

**Comment by pbatard** [comment]
Need a new full log from your new attempt or any attempt that fails and that you want to report.

I simply cannot help you with a partial report that does not contain the full log from Rufus.

**Comment by Sixhackers** [comment]
Pete;
Attached is the full log of the failure with the message "Unable to assign a drive letter"
Yes I now know I screwed up trying to use it stored on the C drive. But the HD I had tried to upgrade to Win 11became unbootable. So installed it in an external HD case to re-partition and re-clone it from my backup HD and start again. So I deleted all partitions on the drive and reformatted it to NTFS using EaseUS Partition Master V15. But it said the partition was locked and would take a two step procedure to unlock the drive and format it after a restart. That completed fine but never saw that needed step before?? Then I cloned it again using EaseUS Backup Workstation V 8.9.
   So something happened with my mistake that "LOCKED the C Drive" when I mistakenly tried to use a stored Windows 11 image on the C drive about to be re-imaged. So after the successful re-partition and re-clone procedures I tried the upgrade again and got the same FAILED "Unable to assign a drive letter message again. And YES the Win 11 image was now on a USB A external flash memory.  So I'm amazed as the very 1st time I tried this on this HD a week ago it worked fine and upgraded to Win 11 when I did it correctly with the Win 11 image on an external drive.
  But when after many Win 11 updates the last update cause the glitchy Internet connection as described in the known issues section. That's when I re-cloned my Win 10 image and AGAIN created this same error issue with your now correct directions.
  I tell you all this as I'm amazed that after the clone the HD again it gave the same error on a freshly cloned image USING YOUR CORRECT procedure, So I'm hoping the attached log will help you to figure out what I broke on this HD that a re-cloned image still can't correct. NOTE: I had run Chkdsk /f /r /x, and DISM /online /cleanup-image /restorehealth BEFOR I tried the upgrade again and they completed and didn't report any errors. BUT I had not run sfc /scannow which I just did after the last failure which created the attached log. So because  sfc /scannow DID find errors after the new clone failure perhaps this HD will again upgrade correctly. The sfc /scannow command also created a CBS log file named Windir/logs/cbs/cbs.log. But I'l be damned if I can find where it put it on the HD! So I'l have win Explorer search the whole re-repaired HD and if I can find the cbs.log file by morning I'l attach it to another reply. Maybe it too may help figure out what I did to this HD that would now make it report the same error with the Win 11 image correctly stored on an external USB drive. Perhaps others will make this same mistake and also need your help to correct the error it creates. I'm hoping the sfc /scannow was the HD fix and hopefully it will upgrade again on the next try. I'l try to find and send the second SFC cbs.log first; and try the upgrade in the morning.
  Please forgive my long explanation, as I'm hoping it helps you or I figure out what's again causing this same dam error again even after a re-clone. Perhaps all I needed was the sfc /scannow command run after cloning it but I'm lost as to why my fresh cloned Win 10 image failed again when now done per your EXACT directions. Attached is the Rufus log. I send the cbs log in the morning if I can find it. Thanks again for all your help and such a great project you've supplied to us all.
Bill Nelson     ***@***.***

________________________________

**Comment by pbatard** [comment]
Attachments do not work when replying to e-mails. You need to go to https://github.com/pbatard/rufus/issues/2606 and edit your content there if you want to provide a log, as the only data I get is the one everybody can see at https://github.com/pbatard/rufus/issues/2606, and your log is **NOT** attached there.

Also, I'm sorry, but while the issue and the steps you took as well as the reason why you got there, may be clear in your head, your report of it is extremely confusing (and poorly formatted if I might add), which makes it even more unlikely that you will find help.

Again, I can not help you unless I see a full log, because your explanations are just too confusing for someone who has no idea about your environment.

**Comment by Sixhackers** [comment]
[rufus.log](https://github.com/user-attachments/files/17720494/rufus.log)
Pete;
Sorry for the cofusing email. Please let me try again and try to simplify the description.
I originally used Rufus V 4.6 correctly with the Win 11.iso on an external drive and it worked and installed Win 11. But after many Win 11 updates it started to do that known issue of a intermittant wifi connection. So after a while I gave up and deleted the imasge, created a new NTFS partition and then re-cloned the HD from my master back up hard drive. From that point after any attempt to upgrade it again to Win 11 causes the failure and the message "Unable to assign a drive letter". I've read the (Attached FULL LOG) but I can assure you the Win11.is0 is ONLY on the extermal memory drive, NOT on the C; drive. And any Win11.iso file that I deleted from the C drive was also deleted from the recycle bin. I'm lost at this point. Any help appreciated but if to confusing just forget it and I'l live out the last year and save for a new Win11 machine. Thanks for your help. You've written a great program. I screwed it up by having an image on the main C drive my 2nd try. I know better now.That's been removed but the upgrade still alway fails with "Unable to assign a drive letter Thanks Bill Nelson sixhackers@hotmail.com

**Comment by Sixhackers** [comment]
Pete;
I've done that. The Win11.iso image is on an external drive not on the C drive! I don't know where else I could possibly put it. So very sorry you don't understand this no matter how many times I say it.  So please forget it. I quit, and won't bother you any longer. Thanks for trying ...Bill
________________________________

**Comment by JonnyTech** [comment]
@Sixhackers:

Rufus will wipe the external drive and copy the contents of the iso to it.

Put the iso and rufus on your hard drive (c:).

Insert the external drive that is to be wiped and formatted into a usb port.

Run rufus from the hard drive c: and select the external drive as the destination.

Rufus will wipe and format the external drive then copy the contents of the iso to the drive.

That is why you should have nothing on your external drive.
